### ISSUE TITLE
Example Playbook

### ISSUE THREAD

**Issue by mira-miracoli** [Paprikant/ansible-role-beacon#1]
Your role looks pretty good I think.
I have trouble with the molecule tests, they don't work for me. (Systemd doesnt work in my containers with your configuration, I suggest switching to podman)
I ran your example playbook against a VM and the playbook ran through, but when I try to reproduce your verify playbook, it fails (also when doing manually) because of this:
```
bash-5.1# beacon_init --min_allele_count 0 /example.vcf /example_metadata.json
[2022-10-24 14:41:57,021][beacon][32 MainProcess][INFO    ] (L:61) db_load | __init__: Start database routines
[2022-10-24 14:41:57,021][beacon][32 MainProcess][INFO    ] (L:180) db_load | connection: Establish a connection to database
[2022-10-24 14:41:57,028][beacon][32 MainProcess][ERROR   ] (L:191) db_load | connection: AN ERROR OCCURRED WHILE ATTEMPTING TO CONNECT TO DATABASE -> p
assword authentication failed for user "beacon"
[2022-10-24 14:41:57,028][beacon][32 MainProcess][INFO    ] (L:195) db_load | check_tables: Request tables from database
Traceback (most recent call last):
  File "/usr/local/bin/beacon_init", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/beacon_api/utils/db_load.py", line 455, in main
    asyncio.run(init_beacon_db())
  File "/usr/local/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.8/site-packages/beacon_api/utils/db_load.py", line 410, in init_beacon_db
    tables = await db.check_tables(["beacon_dataset_table", "beacon_data_table", "beacon_dataset_counts_table"])
  File "/usr/local/lib/python3.8/site-packages/beacon_api/utils/db_load.py", line 197, in check_tables
    tables = await self._conn.fetch(
AttributeError: 'NoneType' object has no attribute 'fetch'
```
The `curl localhost:5050` returns 500

My playbook looks like this:
```yaml
- hosts: all
  become: true
  tasks:
    - name: "Include paprikant.beacon"
      include_role:
        name: "paprikant.beacon"
      vars:
        postgres_pass: myverysecretpass
```

**Comment by Paprikant** [comment]
That does not sound like it worked for you :laughing: 

I used docker for the molecule deployments to easily [run](https://github.com/Paprikant/ansible-role-beacon/actions/runs/3280204648/jobs/5400700475) it on githubs workers (ubuntu with preinstalled docker). Maybe ill change it to podman in the future.

Your playbook works just fine for me. Is there any chance you changed the postgres password at some point and re-run the playbook?
That would lead to the outcome you are describing (The database was created with one user, then beacon-python is changed to use another one - the postgres user however remains)

I already left a little yellow warning sign in the Readme about this.

**Comment by mira-miracoli** [comment]
I changed the password, yes, because it contained a german character, the application did not like.
But, I deleted both containers before I re-run the playbook.
I can also try with a new VM in that case.
Is there a possibility to change the ansible role so that a new password would be deployed to the beacon container, too?

**Comment by Paprikant** [comment]
Hmm maybe it would be a good idea to make a locale configurable (postgres uses `en_US.utf8` by default) to avoid problems with special characters.

The actual database files live in a host bind at `postgres_data_dir` (by default /var/lib/postgresql/data). Therefore removing the containers did not have an impact on the existing database (maybe the README is not clear here - it just says "where to store database files" :sweat_smile: ).

User creation happens during database initialization which is run when there exists no database (i.e. empty  `postgres_data_dir`).
This is something you can bring about with `rm -rf var/lib/postgresql/data/*`before running the playbook with a changed password (the container will be recreated automatically since an environment variable has changed)

In the playbook we do not want to do that since it would drop the database every time the playbook is run :laughing: 

--- 

In order to change the password on an existing deployment we would want to execute something along the lines of

```
CREATE USER <new user> PASSWORD '<new password>'`
GRANT ALL PRIVILEGES ON *.* TO <new user> WITH GRANT OPTION
DROP USER <old username>
```

Therefore we would need access to the database -> therefore we need the old credentials -> which are now no longer in the `postgres_user` and `postgres_pass` variable since we just changed them.

I am not sure how to best keep the old credentials accessible since I never worked with ansible. The best way I can think of is to store the current user and password in  a file locally (on the host), and read this file when the playbook is run on a host that already has the file.

But I did not like the idea of leaving the plain text database password somewhere in the file system. This is the reason I just left a warning in the Readme and didnt address password changes at all.


TL;DR
It is possible to change the password automatically, but the solution i came up with didnt feel right to me.

**Comment by mira-miracoli** [comment]
And to store things safely in ansible, I would recomend using ansible-vault

**Comment by Paprikant** [comment]
Good suggestion on the geerlingguy :) https://github.com/Paprikant/ansible-role-beacon/pull/2
